code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -4467,3 +4469,275 @@
+	description = "Not safe for children above or under the age of 12."
+
+/datum/reagent/drink/tea/mint
+	name = "Groans Tea: Minty Delight Flavor"
+	id = "mint"
+	description = "Very filling!"
+
+/datum/reagent/drink/tea/chamomile
+	name = "Groans Tea: Chamomile Flavor"
+	id = "chamomile"
+	description = "Enjoy a good night's sleep."
+
+/datum/reagent/drink/tea/exchamomile
+	name = "Tea"
+	id = "exchamomile"
+	description = "Who needs to wake up anyway?"
+
+/datum/reagent/drink/tea/fancydan
+	name = "Groans Banned Tea: Fancy Dan Flavor"
+	id = "fancydan"
+	description = "Full of that patented Dan taste you love!"
+
+/datum/reagent/drink/tea/plasma
+	name = "Plasma Pekoe"
+	id = "plasma"
+	description = "Probably not the safest beverage."
+
+/datum/reagent/drink/hot_coco
+	name = "Hot Chocolate"
+	id = "hot_coco"
+	description = "Made with love! And coco beans."
+	nutriment_factor = 2 * FOOD_METABOLISM
+	color = "#403010" // rgb: 64, 48, 16
+	adj_temp = 5
+
+/datum/reagent/drink/coffee
+	name = "Coffee"
+	id = "coffee"
+	description = "Coffee is a brewed drink prepared from roasted seeds, commonly called coffee beans, of the coffee plant."
+	color = "#482000" // rgb: 72, 32, 0
+	adj_dizzy = -5
+	adj_drowsy = -3
+	adj_sleepy = -2
+	adj_temp = 25
+
+/datum/reagent/drink/coffee/on_mob_life(var/mob/living/M as mob)
+
+	if(!holder) return
+	..()
+	if(!holder)
+		holder = M.reagents
+	if(holder)
+		M.Jitter(5)
+		if(adj_temp > 0 && holder.has_reagent("frostoil"))
+			holder.remove_reagent("frostoil", 10*REAGENTS_METABOLISM)
+
+		holder.remove_reagent(src.id, 0.1)
+
+/datum/reagent/drink/coffee/icecoffee
+	name = "Iced Coffee"
+	id = "icecoffee"
+	description = "Coffee and ice, refreshing and cool."
+	color = "#102838" // rgb: 16, 40, 56
+	adj_temp = -5
+
+/datum/reagent/drink/coffee/soy_latte
+	name = "Soy Latte"
+	id = "soylatte"
+	description = "A nice and tasty beverage while you are reading your hippie books."
+	color = "#664300" // rgb: 102, 67, 0
+	adj_sleepy = 0
+	adj_temp = 5
+
+/datum/reagent/drink/coffee/soy_latte/on_mob_life(var/mob/living/M as mob)
+		..()
+		M.sleeping = 0
+		if(M.getBruteLoss() && prob(20)) M.heal_organ_damage(1,0)
+		return
+
+/datum/reagent/drink/coffee/latte
+	name = "Latte"
+	id = "latte"
+	description = "A nice, strong and tasty beverage while you are reading."
+	color = "#664300" // rgb: 102, 67, 0
+	adj_sleepy = 0
+	adj_temp = 5
+
+/datum/reagent/drink/coffee/latte/on_mob_life(var/mob/living/M as mob)
+
+	if(!holder) return
+	..()
+	M.sleeping = 0
+	if(M.getBruteLoss() && prob(20)) M.heal_organ_damage(1,0)
+	return
+
+/datum/reagent/drink/coffee/espresso
+	name = "Espresso"
+	id = "espresso"
+	description = "Coffee made with water."
+
+//Let's hope this one works
+var/global/list/tonio_doesnt_remove=list(
+	"tonio",
+	"blood"
+)
+
+
+/datum/reagent/drink/coffee/tonio
+	name = "Tonio"
+	id = "tonio"
+	nutriment_factor = 1 * FOOD_METABOLISM
+
+
+
+/datum/reagent/tonio/on_mob_life(var/mob/living/M as mob)
```

Comment:
See the comment above as to pathing.

If this is an effect also present in charcoal and chifir, you should make it a general proc.